network-api/networkapi/campaign/views.py

```diff
@@ -52,10 +52,15 @@ def send_message(self, QueueUrl, MessageBody):
 logger = logging.getLogger(__name__)
 
 
-@api_view(['POST'])
-@parser_classes((JSONParser,))
-@permission_classes((permissions.AllowAny,))
+@csrf_exempt
+@require_http_methods(['POST'])
 def signup_submission_view(request, pk):
+    # We need to re-write the data that's coming in from the XMLHttpRequest.
+    # XMLHttpRequest's send data through the request.body, not request.POST despite it being a POST method
+    # request.POST is supported for unit tests
```

Comment:
`XMLHttpRequest` was replaced by the Fetch API a long time ago (in web years), let's call this a "network request" instead.

Reply:
It's _actually_ using an XmlHttpRequest in the JavaScript though. But I can update, Im not picky in that.